include/secp256k1_musig.h

```diff
@@ -7,348 +7,152 @@
- *  `musig_partial_signature_serialize` and `musig_partial_signature_parse`
- *  functions.
- */
+/** Opaque data structure that holds a signer's public nonce.
+*
+*  Guaranteed to be 132 bytes in size. Serialized and parsed with
+*  `musig_pubnonce_serialize` and `musig_pubnonce_parse`.
+*/
 typedef struct {
-    unsigned char data[32];
-} secp256k1_musig_partial_signature;
+    unsigned char data[132];
+} secp256k1_musig_pubnonce;
 
-/** Computes a combined public key and the hash of the given public keys.
- *
- *  Different orders of `pubkeys` result in different `combined_pk`s.
- *
- *  The pubkeys can be sorted before combining with `secp256k1_xonly_sort` which
- *  ensures the same resulting `combined_pk` for the same multiset of pubkeys.
- *  This is useful to do before pubkey_combine, such that the order of pubkeys
- *  does not affect the combined public key.
+/** Opaque data structure that holds an aggregate public nonce.
  *
- *  Returns: 1 if the public keys were successfully combined, 0 otherwise
- *  Args:        ctx: pointer to a context object initialized for verification
- *                    (cannot be NULL)
- *           scratch: scratch space used to compute the combined pubkey by
- *                    multiexponentiation. If NULL, an inefficient algorithm is used.
- *  Out: combined_pk: the MuSig-combined xonly public key (cannot be NULL)
- *       pre_session: if non-NULL, pointer to a musig_pre_session struct to be used in
- *                    `musig_session_init` or `musig_pubkey_tweak_add`.
- *   In:     pubkeys: input array of pointers to public keys to combine. The order
- *                    is important; a different order will result in a different
- *                    combined public key (cannot be NULL)
- *         n_pubkeys: length of pubkeys array. Must be greater than 0.
+ *  Guaranteed to be 132 bytes in size. Serialized and parsed with
+ *  `musig_aggnonce_serialize` and `musig_aggnonce_parse`.
  */
-SECP256K1_API int secp256k1_musig_pubkey_combine(
-    const secp256k1_context* ctx,
-    secp256k1_scratch_space *scratch,
-    secp256k1_xonly_pubkey *combined_pk,
-    secp256k1_musig_pre_session *pre_session,
-    const secp256k1_xonly_pubkey * const* pubkeys,
-    size_t n_pubkeys
-) SECP256K1_ARG_NONNULL(1) SECP256K1_ARG_NONNULL(3) SECP256K1_ARG_NONNULL(5);
+typedef struct {
+    unsigned char data[132];
+} secp256k1_musig_aggnonce;
 
-/** Tweak an x-only public key by adding the generator multiplied with tweak32
- *  to it. The resulting output_pubkey with the given internal_pubkey and tweak
- *  passes `secp256k1_xonly_pubkey_tweak_test`.
- *
- *  This function is only useful before initializing a signing session. If you
- *  are only computing a public key, but not intending to create a signature for
- *  it, you can just use `secp256k1_xonly_pubkey_tweak_add`. Can only be called
- *  once with a given pre_session.
+/** Opaque data structure that holds a cache for a MuSig session.
```

Comment:
Style nit (feel free to ignore): "holds a cache" maybe makes it sound like an optional performance optimization; should it say something like "holds the state" instead?

Reply:
Removed "a cache for"